GDJS/Runtime/scenestack.ts

```diff
@@ -91,13 +100,33 @@ namespace gdjs {
      * Pause the scene currently being played and start the new scene that is specified.
      * If `externalLayoutName` is set, also instantiate the objects from this external layout.
      */
-    push(newSceneName: string, externalLayoutName?: string): gdjs.RuntimeScene {
+    push(
+      newSceneName: string,
+      externalLayoutName?: string
+    ): gdjs.RuntimeScene | null {
       // Tell the scene it's being paused
       const currentScene = this._stack[this._stack.length - 1];
       if (currentScene) {
         currentScene.onPause();
       }
 
+      if (this._runtimeGame.areLayoutAssetsReady(newSceneName)) {
+        return this._loadNewScene(newSceneName, externalLayoutName);
+      }
+
+      this._nextLayout = newSceneName;
+      this._isNextLayoutLoading = true;
+      this._runtimeGame.loadSceneAssets(newSceneName).then(() => {
+        this._isNextLayoutLoading = false;
```

Comment:
Any reason why `this._loadNewScene(this._nextLayout);` could not be here and is in `step` instead?

Reply:
I was probably afraid that the scene expression would change during events execution but it doesn't use the stack so it doesn't seem necessary indeed.